cooked-validators/src/Cooked/MockChain/Monad/Direct.hs

```diff
@@ -242,15 +243,17 @@ utxoIndex0 = utxoIndex0From def
 -- ** Direct Interpretation of Operations
 
 instance (Monad m) => MonadBlockChain (MockChainT m) where
-  validateTxSkel skel = do
+  validateTxSkel skelUnbal = do
+    (firstSigner : _) <- pure $ toList $ skelUnbal ^. txSkelRequiredSigners
```

Comment:
We could require an explicit balancing wallet (and call it so instead of making it ambiguous like before with `signs` and `as`) when validating a transaction skeleton. E.g. `validateTxSkel (wallet 2) txSkel`.

The `_txSkelRequiredSigners` field in `TxSkel` would be solely used for contract related signing requirements.